modules/python.nix

```diff
@@ -0,0 +1,145 @@
+{ pkgs, ... }:
+let
+  pip = pkgs.callPackage ../pkgs/pip { };
+
+  poetry = pkgs.callPackage ../pkgs/poetry { };
+
+  python = pkgs.python310Full;
+
+  stderred = pkgs.replitPackages.stderred;
+
+  prybar = pkgs.replitPackages.prybar-python310;
+
+  pypkgs = pkgs.python310Packages;
+
+  debugpy = pypkgs.debugpy;
+
+  dapPython = pkgs.replitPackages.dapPython;
+
+  python-lsp-server = pkgs.callPackage ../pkgs/python-lsp-server { };
+
+  python-ld-library-path = pkgs.lib.makeLibraryPath [
+    # Needed for pandas / numpy
+    pkgs.stdenv.cc.cc.lib
+    pkgs.zlib
+    # Needed for pygame
+    pkgs.glib
+    # Needed for matplotlib
+    pkgs.xorg.libX11
+  ];
+
+  python3-wrapper = pkgs.stdenvNoCC.mkDerivation {
+    name = "python3-wrapper";
+    buildInputs = [ pkgs.makeWrapper ];
+
+    buildCommand = ''
+      mkdir -p $out/bin
+      makeWrapper ${python}/bin/python3 $out/bin/python3 \
+        --set LD_LIBRARY_PATH "${python-ld-library-path}"
+    
+      ln -s $out/bin/python3 $out/bin/python
+      ln -s $out/bin/python3 $out/bin/python3.10
```

Comment:
oof, this might bite us in the future: minor versions of python are not mutually-compatible, since sometimes Python maintainers _remove_ old functionality: we may need to call this module `python_3_10`.